### ISSUE TITLE
Maximum call stack exceeded when installing from a private github repo

### ISSUE THREAD

**Issue by gscott** [npm/npm#9160]
If I go back to npm 2.x it works.

Command:
````
npm install git+ssh://git@github.com:<my-org>>/<my-repo> --save
````
npm-debug.log below: if my obfuscation of my org and repo names is too annoying let me know:
````
0 info it worked if it ends with ok
1 verbose cli [ '/usr/local/bin/iojs',
1 verbose cli   '/usr/local/bin/npm',
1 verbose cli   'install',
1 verbose cli   'git+ssh://git@github.com:<my-org>/<my-repo>',
1 verbose cli   '--save' ]
2 info using npm@3.2.1
3 info using node@v2.5.0
4 silly loadCurrentTree Starting
5 silly install loadCurrentTree
6 silly install readLocalPackageData
7 silly fetchPackageMetaData git+ssh://git@github.com:<my-org>/<my-repo>
8 silly fetchOtherPackageData git+ssh://git@github.com:<my-org>/<my-repo>
9 silly cache add args [ 'git+ssh://git@github.com:<my-org>/<my-repo>', null ]
10 verbose cache add spec git+ssh://git@github.com:<my-org>/<my-repo>
11 silly cache add parsed spec { raw: 'git+ssh://git@github.com:<my-org>/<my-repo>',
11 silly cache add   scope: null,
11 silly cache add   name: null,
11 silly cache add   rawSpec: 'git+ssh://git@github.com:<my-org>/<my-repo>',
11 silly cache add   spec: 'git+ssh://git@github.com/<my-org>/<my-repo>.git',
11 silly cache add   type: 'hosted',
11 silly cache add   hosted:
11 silly cache add    { type: 'github',
11 silly cache add      ssh: 'git@github.com:<my-org>/<my-repo>.git',
11 silly cache add      sshUrl: 'git+ssh://git@github.com/<my-org>/<my-repo>.git',
11 silly cache add      httpsUrl: 'git+https://github.com/<my-org>/<my-repo>.git',
11 silly cache add      gitUrl: 'git://github.com/<my-org>/<my-repo>.git',
11 silly cache add      shortcut: 'github:<my-org>/<my-repo>',
11 silly cache add      directUrl: 'https://raw.githubusercontent.com/<my-org>/<my-repo>/master/package.json' } }
12 verbose addRemoteGit caching git+ssh://git@github.com:<my-org>/<my-repo>
13 verbose addRemoteGit git+ssh://git@github.com/<my-org>/<my-repo>.git is a repository hosted by github
14 silly tryClone cloning git+ssh://git@github.com/<my-org>/<my-repo>.git via git+ssh://git@github.com/<my-org>/<my-repo>.git
15 verbose tryClone git-ssh-git-github-com-<my-org>-<my-repo>-git-58e4d599879d6aaa25014b8ff66efd16 not in flight; caching
16 info git [ 'clone',
16 info git   '--template=/Users/gs/.npm/_git-remotes/_templates',
16 info git   '--mirror',
16 info git   'ssh://git@github.com/<my-org>/<my-repo>.git',
16 info git   '/Users/gs/.npm/_git-remotes/git-ssh-git-github-com-<my-org>-<my-repo>-git-58e4d599879d6aaa25014b8ff66efd16' ]
17 verbose mirrorRemote git+ssh://git@github.com/<my-org>/<my-repo>.git git clone ssh://git@github.com/<my-org>/<my-repo>.git
18 verbose setPermissions git+ssh://git@github.com/<my-org>/<my-repo>.git set permissions on /Users/gs/.npm/_git-remotes/git-ssh-git-github-com-<my-org>-<my-repo>-git-58e4d599879d6aaa25014b8ff66efd16
19 verbose resolveHead git+ssh://git@github.com/<my-org>/<my-repo>.git original treeish: master
20 info git [ 'rev-list', '-n1', 'master' ]
21 silly resolveHead git+ssh://git@github.com/<my-org>/<my-repo>.git resolved treeish: 178fb884e841c8296fda898bbd7fb6ca293505ff
22 verbose resolveHead git+ssh://git@github.com/<my-org>/<my-repo>.git resolved Git URL: git+ssh://git@github.com/<my-org>/<my-repo>.git#178fb884e841c8296fda898bbd7fb6ca293505ff
23 silly resolveHead Git working directory: /var/folders/vc/27csn2qx2qd6_2mcsbrf3hcndh8xl6/T/npm-95063-bd3aef31/git-cache-52786a15c883209998958867d9995b71/178fb884e841c8296fda898bbd7fb6ca293505ff
24 info git [ 'clone',
24 info git   '/Users/gs/.npm/_git-remotes/git-ssh-git-github-com-<my-org>-<my-repo>-git-58e4d599879d6aaa25014b8ff66efd16',
24 info git   '/var/folders/vc/27csn2qx2qd6_2mcsbrf3hcndh8xl6/T/npm-95063-bd3aef31/git-cache-52786a15c883209998958867d9995b71/178fb884e841c8296fda898bbd7fb6ca293505ff' ]
25 verbose cloneResolved git+ssh://git@github.com/<my-org>/<my-repo>.git clone Cloning into '/var/folders/vc/27csn2qx2qd6_2mcsbrf3hcndh8xl6/T/npm-95063-bd3aef31/git-cache-52786a15c883209998958867d9995b71/178fb884e841c8296fda898bbd7fb6ca293505ff'...
25 verbose cloneResolved done.
26 info git [ 'checkout', '178fb884e841c8296fda898bbd7fb6ca293505ff' ]
27 verbose checkoutTreeish git+ssh://git@github.com/<my-org>/<my-repo>.git checkout Note: checking out '178fb884e841c8296fda898bbd7fb6ca293505ff'.
27 verbose checkoutTreeish
27 verbose checkoutTreeish You are in 'detached HEAD' state. You can look around, make experimental
27 verbose checkoutTreeish changes and commit them, and you can discard any commits you make in this
27 verbose checkoutTreeish state without impacting any branches by performing another checkout.
27 verbose checkoutTreeish
27 verbose checkoutTreeish If you want to create a new branch to retain commits you create, you may
27 verbose checkoutTreeish do so (now or later) by using -b with the checkout command again. Example:
27 verbose checkoutTreeish
27 verbose checkoutTreeish   git checkout -b new_branch_name
27 verbose checkoutTreeish
27 verbose checkoutTreeish HEAD is now at 178fb88... add example log:false to readme
28 verbose addLocalDirectory /Users/gs/.npm/<my-repo>/2.3.1/package.tgz not in flight; packing
29 verbose tar pack [ '/Users/gs/.npm/<my-repo>/2.3.1/package.tgz',
29 verbose tar pack   '/var/folders/vc/27csn2qx2qd6_2mcsbrf3hcndh8xl6/T/npm-95063-bd3aef31/git-cache-52786a15c883209998958867d9995b71/178fb884e841c8296fda898bbd7fb6ca293505ff' ]
30 verbose tarball /Users/gs/.npm/<my-repo>/2.3.1/package.tgz
31 verbose folder /var/folders/vc/27csn2qx2qd6_2mcsbrf3hcndh8xl6/T/npm-95063-bd3aef31/git-cache-52786a15c883209998958867d9995b71/178fb884e841c8296fda898bbd7fb6ca293505ff
32 verbose stack RangeError: Maximum call stack size exceeded
32 verbose stack     at byteLength (buffer.js:288:20)
32 verbose stack     at fromString (buffer.js:65:16)
32 verbose stack     at new Buffer (buffer.js:48:5)
32 verbose stack     at encodeField (/usr/local/lib/node_modules/npm/node_modules/tar/lib/extended-header-writer.js:179:16)
32 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/tar/lib/extended-header-writer.js:139:32
32 verbose stack     at Array.forEach (native)
32 verbose stack     at encodeFields (/usr/local/lib/node_modules/npm/node_modules/tar/lib/extended-header-writer.js:98:23)
32 verbose stack     at /usr/local/lib/node_modules/npm/node_modules/tar/lib/extended-header-writer.js:137:32
32 verbose stack     at Array.forEach (native)
32 verbose stack     at encodeFields (/usr/local/lib/node_modules/npm/node_modules/tar/lib/extended-header-writer.js:98:23)
33 verbose cwd /Users/gs/dev/app-chatbot
34 error Darwin 14.4.0
35 error argv "/usr/local/bin/iojs" "/usr/local/bin/npm" "install" "git+ssh://git@github.com:<my-org>/<my-repo>" "--save"
36 error node v2.5.0
37 error npm  v3.2.1
38 error Maximum call stack size exceeded
39 error If you need help, you may report this error at:
39 error     <https://github.com/npm/npm/issues>
40 verbose exit [ 1, true ]

**Comment by othiym23** [comment]
@iarna: is this a duplicate of #8706 or #8807?

**Comment by iarna** [comment]
@othiym23 Maybe? Not clear yet. I'm using https://github.com/npm/npm/issues/9224 to track the larger issue here.

**Comment by othiym23** [comment]
Closing as a duplicate of #9239 / #9224 / #8706 / #8807 – we've made some changes to the publishing code to guard against this problem (which `npm pack` calls), so we believe this has been fixed as of `npm@3.3.5`. Update your npm and let us know if you're still seeing this issue!

**Comment by gscott** [comment]
npm updated to 3.3.8 and no longer see the issue, thank you
